dashboard/test/models/contact_rollups_pardot_memory_test.rb

```diff
@@ -32,52 +32,167 @@ class ContactRollupsPardotMemoryTest < ActiveSupport::TestCase
     assert_equal new_pardot_id, ContactRollupsPardotMemory.find_by(email: existing_record.email)&.pardot_id
   end
 
-  test 'create_new_pardot_contacts' do
+  test 'query_new_contacts' do
     assert_equal 0, ContactRollupsPardotMemory.count
     assert_equal 0, ContactRollupsProcessed.count
-    PardotV2.stubs(:submit_batch_request).returns([])
 
-    # Sync 1 contact
-    processed_contacts = []
-    processed_contacts << (create :contact_rollups_processed)
+    pardot_memory_records = [
+      {email: 'alpha', pardot_id: nil},
+      {email: 'beta', pardot_id: 1}
+    ]
+    pardot_memory_records.each {|record| create :contact_rollups_pardot_memory, record}
 
-    ContactRollupsPardotMemory.create_new_pardot_prospects
-    assert_equal processed_contacts.length, ContactRollupsPardotMemory.count
+    processed_contact_records = [
+      {email: 'alpha'}, {email: 'beta'}, {email: 'gamma'}
+    ]
+    processed_contact_records.each {|record| create :contact_rollups_processed, record}
+
+    # Execute SQL query
+    results = ActiveRecord::Base.connection.
+      exec_query(ContactRollupsPardotMemory.query_new_contacts).map do |record|
+      record['email']
+    end
+
+    # Should find only 2 new contacts
+    assert_equal %w(alpha gamma), results
```

Comment:
do we not find beta because it had a pardot_id initially? could you add a comment explaining that?